### PR TITLE
Patch airflow 1.10.1

### PR DESCRIPTION
**Pull request by seeholza**
- removed deprecation warning by passing context always (as in triggerdagrun)
- using timezone instead of datetime

### CODE REVIEW

File: multi_dagrun/operators/multi_dagrun.py [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
         self.op_args = op_args or []
         self.op_kwargs = op_kwargs or {}
-        self.provide_context = provide_context
 
     def execute(self, context):
-        if self.provide_context:
-            context.update(self.op_kwargs)
-            self.op_kwargs = context
-
+        context.update(self.op_kwargs)
```
Comment (mastak) [inline multi_dagrun/operators/multi_dagrun.py:20]: It's a not good idea to made updates to a mutable incoming argument.

File: examples/simple_trigger.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 from airflow.utils.dates import days_ago
 
 
-def generate_dag_run():
+def generate_dag_run(**kwargs):
```
Comment (mastak) [inline examples/simple_trigger.py:6]: Unused variable looks really strange. As for me, it looks like some bug in the code.

**Review by mastak — CHANGES_REQUESTED** [review]

**Comment by mastak** [comment]
Hi @flinz,

Thanks for PR. I totally agree about `timezone`, but I'm not sure about 
```removed deprecation warning by passing context always (as in triggerdagrun)```
I have run the code but didn't see any warnings.

Right now I don't see any requirements to pass context to the function all times. Do you have opposite arguments for it?

I know that `TriggerDagRunOperator` pass context always, but for the `PythonOperator` way more preferred where you can tell to pass context or not.

**Comment by seeholza** [comment]
> Right now I don't see any requirements to pass context to the function all times. Do you have opposite arguments for it?
> I know that TriggerDagRunOperator pass context always, but for the PythonOperator way more preferred where you can tell to pass context or not.

Since it is inherited from TriggerDagRunOperator, I was under the impression it would be more consistent to have it behave the same way. But I tend to agree that then op_args and op_kwargs would also have to be removed. since these are useful, and I we could leave it as is.

> I have run the code but didn't see any warnings.

Weird. I have looked at it again, it might have been an error on my side.. I added this back in, due to the above reasoning.

File: examples/simple_trigger.py (hunk 2 above)
Comment (seeholza) [reply examples/simple_trigger.py:6]: Agree, but this was necessary when passing the context always. Resolved by the updates.

File: multi_dagrun/operators/multi_dagrun.py (hunk 1 above)
Comment (seeholza) [reply multi_dagrun/operators/multi_dagrun.py:20]: Noted. Switched back to the original formulation

**Comment by seeholza** [comment]
Ok, I see you patched the one thing necessary. I will close the PR.

**Pull request closed without merging**